zebra-chain/src/transaction/hash.rs

```diff
@@ -13,7 +13,7 @@ use super::Transaction;
 ///
 /// TODO: I'm pretty sure this is also a SHA256d hash but I haven't
 /// confirmed it yet.
-#[derive(Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Copy, Clone, Eq, PartialEq, Serialize, Deserialize, Hash)]
```

Comment:
A `Hash` is the result of something that is `Hash`able, ie, the result of something calling `Hash.hash()` like say a `Transaction`, so I think this trait should be removed (also I don't think it's being used anywhere)

Reply:
I'm not really sure how best to avoid this, we could use a `BTreeMap` instead of a hashmap to avoid the Hash bound but I don't think we can use the fact that this is a hash to implement the `Hash` trait.

That said, whats the issue with hashing a Hash? Is it just runtime cost?

Reply:
Ah I forgot you need to impl `Hash` to be a key in the `HashMap`, ok ignore me then. 
I might add a comment here just to be like: 'this Hash impls Hash because it's being used as a key in the HashMap's in InventoryRegistry'

Reply:
I think it's fine for types to implement `Hash` without explanation, it's a good practice for public types to implement `Hash` and as many of the derivable traits as makes sense for their use-case, just so that caller code is as flexible as possible.  Because derives have a different namespace there's no risk of a name collision.

Reply:
Yeah I'm not so much worried about name collision as the thing that happened to me, which is, 'why is this type that is the result of hashing something, also being hashed?', the answer is _our_ `transaction::Hash` is 'business data model' and the `Hash` trait and use of it is implementation detail of `InventoryRegistry`, humans just have to remember all the levels of what's going on, and a comment can help clarify that. "We aren't just implementing `Hash` because this type is called `Hash`..." etc